courses/sync_external_courses/utils.py

```diff
@@ -0,0 +1,202 @@
+"""
+Statistics collection and reporting for external course synchronization.
+
+This module provides classes for collecting, organizing, and reporting
+statistics about external course synchronization operations, with a focus
+on named properties for better readability and maintenance.
+"""
+
+from dataclasses import dataclass
+from typing import Optional
+
+
+@dataclass
+class CoursewareInfo:
+    """
+    Data class for courseware information with named fields
+    """
+
+    code: str
+    title: Optional[str] = None
+    msg: Optional[str] = None
+
+    def __eq__(self, other):
+        """
+        Equality based solely on the code field.
+        This allows sets to consider objects with the same code as equal.
+        """
+        if not isinstance(other, CoursewareInfo):
+            return False
+        return self.code == other.code
+
+    def __hash__(self):
+        """
+        Hash based solely on the code field.
+        This is needed for set operations and dictionary keys.
+        """
+        return hash(self.code)
+
+
+class StatItemsCollection:
+    """
+    Manages a collection of statistical items within a specific category.
+
+    This class serves as both a container for items in a statistics category
+    and a metadata handler that provides display information about the category
+    itself.
```

Comment:
Lets use the same words to avoid confusion.
```suggestion
    Manages a collection of stat items within a specific category.

    This class serves as both a container for items in a stat category
    and a metadata handler that provides display information about the category
    itself.
```